pkg/controller/etcdproxy/certs.go

```diff
@@ -17,6 +18,104 @@ import (
 // ProxyCertificateExpiryAnnotation is annotating are certificates successfully created and stored in the Secret or ConfigMap.
 const ProxyCertificateExpiryAnnotation = "etcd.xmudrii.com/certificate-generated"
 
+// appendClientCACert appends new CA certificate to the Client CA bundle located in the controller ConfigMap.
+func (c *EtcdProxyController) appendClientCACert(etcdstorage *etcdstoragev1alpha1.EtcdStorage, caCert *certs.Certificate) error {
```

Comment:
I expected a helper that would take an existing set of ca bundles as bytes (that's your input from the configmap) and a certs.Certificate and produce a new certs.Certificate with only valid CAs.  That method would then be re-used by both serving and client CA bundle methods